syntax = "proto3";

service Testes {
  rpc ChamadaVazio (Vazio) returns (Vazio);
  rpc ValorAbsolutoLong (Long) returns (Long);
  rpc SomaListaLong (ListaLong) returns (Long);
  rpc InverteString (String) returns (String);
  rpc AdicionaContato (Contato) returns (StatusTransacao);
  rpc PegaContato(ContatoId) returns (Contato);
  rpc AtualizaContato (Contato) returns (StatusTransacao);
  rpc RemoveContato(ContatoId) returns (StatusTransacao);
  rpc ListaContatos(Vazio) returns (Contatos);
}

message Vazio {}

message Long {
  int64 valor = 1;
}

message ListaLong {
  repeated int64 valores = 1;
}

message String {
  string texto = 1;
}

message Contato {
  int64 id = 1;
  string nome = 2;
  string email = 3;
  string telefone = 4;
  repeated string rotulos = 5;

  message Endereco {
    string nomeendereco = 1;
    string cidade = 2;
    string estado = 3;
    string cep = 4;
    string pais = 5;
  }

  Endereco endereco = 6;
  bool favorito = 7;
}

message StatusTransacao {
  bool status = 1;
  string msgstatus = 2;
}

message ContatoId {
  int64 id = 1;
}

message Contatos {
  repeated Contato contatos = 1;
}